### ISSUE TITLE
New bundle is downloaded, but app does not update on staging environment

### ISSUE THREAD

**Issue by gabrielrolfsen** [Microsoft/react-native-code-push#1455]
### Steps to Reproduce
1. Make some visible change (preferably an UI change)
2. Run `appcenter codepush release-react --mandatory`
3. Run targeted version on simulator

### Expected Behavior
Version downloaded and changes visible.

### Actual Behavior
First, we got a log telling that the version was successfully downloaded.
```
[CodePush] Reporting CodePush update success (v65)
```
Then, the app is restarted and bundled is successfully downloaded, showing not even a single change made on the codepush version.

This happens on debug and release versions.
![image](https://user-images.githubusercontent.com/4337134/48733131-cf907d00-ec28-11e8-9eb1-e8c77805bf21.png)

### Environment

* react-native-code-push version: 5.4.2
* react-native version: 0.57.1
* iOS/Android/Windows version: iOS 12.0 (16A366)
* Does this reproduce on a debug build or release build? Both
* Does this reproduce on a simulator, or only on a physical device? Simulator

**Comment by alexandergoncharov** [comment]
Hi @gabrielrolfsen,
Thanks for reporting!

Could you please provide full logs and detailed reprosteps or dummy project with reproducing issue? as I can't reproduce this issue, all works fine for me.

Thanks,
Alexander

**Comment by gabrielrolfsen** [comment]
Unfortunately I don't think I have further useful logs to share...
I haven't setup a dummy project to test it yet, but I could test it on my device and I found out that it works, but only for the latest deployed version. When I make UI changes to a previous state of the master branch (an old commit), targeting an older version, it doesn't work.

Any thoughts?

**Comment by alexandergoncharov** [comment]
hmm, so, you changed some UI part, made release and update was successfully installed. After that you made one more changes, made new release and update was not successfully installed. Is it correct? 
If yes, could you please share this changes?

**Comment by gabrielrolfsen** [comment]
Yesterday I deployed to Production environment and tested and it was OK. Unfortunately I could not find a way to test my CodePushes on Staging environment.
I will try to describe what happened in steps:
Let's assume there's two versions: current (v2.0) and old (v1.0)

1. Checkout an old branch, with an old version of the app (v1.0)
2. Created a new branch, added some UI changes and committed to this new branch.
3. Ran the `appcenter codepush release-react --mandatory`
3.1 Checked Appcenter page, codepush was sent to server successfully
4. Checkout again to the old branch (v1.0)
5. Build it and ran it (yarn start and build on xcode) on a simulator OR a device
6. Check logs
6.1 Log shows the message: [CodePush] Reporting CodePush update success (v65)
7. App is restarted and, apparently patch was installed
7.1 Checked Appcenter page, codepush version install and download counter was incremented
8. Cannot see any of the changes
---

I guess what happens is that even though it downloads and installs the version it keeps sync with my local repository, which has the old version (v1.0). Am I right?
This hypothesis is due to the correct functionality being shown on the production environment, after promoting the codepush vesion without being able to test it (risky, but it was my last resort).
